### ISSUE TITLE
bulk gene add issues

### ISSUE THREAD

**Issue by bkmartinjr** [chanzuckerberg/cellxgene#579]
Several UI/UX issues with current implementation.  Suggest we need to do a better job of parsing and handling text input on gene names.

- [x] UI doesn't accept space separated genes, eg entering "Vim Zeb1" generates an error toast, whereas "Vim, Zeb1" does not.
- [ ] Error toasts are persistent (never go away), which means the user must mouse over and manually dismiss them.  Super annoying (to me anyway) - see image below.
- [x] Double commas cause error on input, eg, "Zim,, Veb1", as does a comma prefix or suffix, eg, ",Vim", "Vim,"
- [x] Hitting return with no text in the input field generates error toast

![image.png](https://images.zenhubusercontent.com/5af38d3b4b5806bc2bd28ea9/bd2c4a1c-95a6-44ed-a4fb-5120e3b57b01)

**Comment by bkmartinjr** [comment]
@neuromusic - may want to consider these for 0.6.

**Comment by neuromusic** [comment]
thanks @bkmartinjr, I'll take a closer look this afternoon

**Comment by colinmegill** [comment]
These issues were fixed as of PR #580, with the exception of dismissing the toasts. I think it's a feature that they stay, since if you add 200 and 30 error, you need to manually figure out which ones they were. If they disappear and you still have 170 in the left hand side, that's a problem IMO. 

My vote is `wontfix` on the remaining issue here, and we close it, unless anyone else has a better suggestion for that - I consider it to be a core workflow issue for scientists.

**Comment by bkmartinjr** [comment]
I defer to the more UI literate on the toast dismiss issue.   I think long term, it suggests we might need a better bulk-add UI, if users add "200".  If they typically add O(10), the toasts seem great.

TL;DR - I'm fine closing this.

**Comment by colinmegill** [comment]
Agree we will need to reevaluate bulk add for 200 genes.
